src/gimelstudio/core/node/node.py

```diff
@@ -85,6 +95,10 @@ def IsOutputNode(self):
     def IsNodeCacheEnabled(self):
         return self._cache_enabled
 
+    def SetPropMute(self, is_muted):
+        for prop in self._properties:   # Could be done differently maybe ?
+            self._properties[prop].SetMute(is_muted)
```

Comment:
What you've done is fine. :slightly_smiling_face:

If you want to other possible solutions, then take a look at [this](https://realpython.com/iterate-through-dictionary-python/) article